data/prepare_dataset.py

```diff
@@ -0,0 +1,90 @@
+import os
+
+import fortepyan as ff
+from tqdm import tqdm
+from quantizer import MidiQuantizer
+from datasets import Value, Dataset, Features, Sequence, DatasetDict, load_dataset
+
+
+def process_dataset(dataset_path: str, split: str, sequence_len: int, quantizer: MidiQuantizer) -> list[dict]:
+    dataset = load_dataset(dataset_path, split=split)
+
+    processed_records = []
+
+    for record in tqdm(dataset, total=dataset.num_rows):
+        # print(record)
+        piece = ff.MidiPiece.from_huggingface(record)
+        processed_record = process_record(piece, sequence_len, quantizer)
+
+        processed_records += processed_record
+
+    return processed_records
+
+
+def process_record(piece: ff.MidiPiece, sequence_len: int, quantizer: MidiQuantizer) -> list[dict]:
+    piece_quantized = quantizer.quantize_piece(piece)
+
+    midi_filename = piece_quantized.source["midi_filename"]
+
+    record = []
+
+    for subset in tqdm(piece_quantized.df.rolling(window=sequence_len, step=sequence_len)):
```

Comment:
```suggestion
    step = sequence_len // 2
    window_split = piece_quantized.df.rolling(window=sequence_len, step=step)
    for subset in tqdm(window_split, leave=False):
```

You can have nested progress bars with `leave=False`